pandas/core/arrays/interval.py

```diff
@@ -543,7 +543,11 @@ def __setitem__(self, key, value):
                 msg = f"'value' should be an interval type, got {type(value)} instead."
                 raise TypeError(msg) from err
 
+        if needs_float_conversion:
```

Comment:
Since we're raising here we can remove the future references that are no longer being used:

https://github.com/pandas-dev/pandas/blob/23b6b935c6ad7f2f24adeb8379d8524c1404dc1f/pandas/core/arrays/interval.py#L554-L555

https://github.com/pandas-dev/pandas/blob/23b6b935c6ad7f2f24adeb8379d8524c1404dc1f/pandas/core/arrays/interval.py#L560-L561